src/query/json.rs

```diff
@@ -126,9 +139,15 @@ impl FromQueryResult for JsonValue {
                     match_postgres_type!(rust_decimal::Decimal);
                     #[cfg(feature = "with-json")]
                     try_get_type!(serde_json::Value, col);
+                    #[cfg(all(feature = "with-json", feature = "postgres-array"))]
+                    try_get_type!(Vec<serde_json::Value>, col);
```

Comment:
Since postgres also supports arrays of json or jsonb, we need to add explicit matching for it in addition to primitive arrays.